src/index.js

```diff
@@ -82,8 +88,36 @@ async function executeActions(params) {
   });
 
   try {
+    // start the redirect process
+    const redirectPromise = redirect(params, ow);
+
+    // start the fetching processes
+    const responsePromise = resolvePreferred(fetchers(params, log).map(invoker));
+
+    const { type, target } = await redirectPromise;
+
+    if (type === 'temporary' || type === 'permanent') {
+      log.info(`${type} redirect to ${target}`);
+      return sendRedirect(redirectPromise);
+    } else if (type === 'internal' && target) {
+      // increase the internal redirect counter
+      const redirects = (params.redirects || 0) + 1;
+      if (redirects > MAX_REDIRECTS) {
+        log.warn(`${type} redirect to ${target} exceeds redirect counter`);
+        return abortRedirect(redirectPromise);
```

Comment:
same as above....I would just pass in the `result`

Reply:
This is a bit different, as the only thing needed is the target, but the response is entirely new.